python-sdk/src/langwatch/telemetry/tracing.py

```
@@ -709,7 +714,12 @@ def trace(
     It can contain multiple spans representing different operations within that cycle.
     """
     # Ensure client is setup
-    ensure_setup(api_key=api_key)
+    ensure_setup()
+    if api_key is not None:
+        warn(
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)